js/src/utils/recordEvent.js

```diff
@@ -119,6 +119,46 @@ export const recordTablePageEvent = ( context, page, direction ) => {
  * @property {string | undefined} trigger (**Deprecated: this property should not be used after Q1 2024**) Action (e.g. `click`)
  */
 
+/**
+ * Triggered when moving to another step during creating/editing a campaign.
+ *
+ * @event gla_paid_campaign_step
+ * @property {string} triggered_by Indicates which button triggered this event
+ * @property {string} action User's action or/and objective (e.g. `go-to-step-2`)
+ * @property {string | undefined} context Indicates where this event happened
+ */
+
+/**
+ * Records the event tracking when calling back the "onClick" of <Stepper> to move to another step.
+ *
+ * @param {string} eventName The event name to record.
+ * @param {string} to The next step to go to, e.g. '2'.
+ * @param {string} [context] Indicates where this event happened.
+ */
+export function recordStepperChangeEvent( eventName, to, context ) {
+	recordEvent( eventName, {
+		triggered_by: `stepper-step${ to }-button`,
+		action: `go-to-step${ to }`,
+		context,
+	} );
+}
```

Comment:
I guess the need to record the starting step would depend on the users of tracking data. This newly added event tracking was meant to align with the event properties of the other two:
- [gla_setup_mc](https://github.com/woocommerce/google-listings-and-ads/blob/11f378c8fb2f3491251fbe35d439b2eb73261937/src/Tracking/README.md#gla_setup_mc)
- [gla_setup_ads](https://github.com/woocommerce/google-listings-and-ads/blob/11f378c8fb2f3491251fbe35d439b2eb73261937/src/Tracking/README.md#gla_setup_ads)

If necessary, it would be a bit better to adjust the three event tracking together in a separate PR.

Reply:
Thanks for the explanation. From my perspective, having knowledge of the initial step would provide a bit more context, but I'm not someone who regularly uses tracking data.